packages/rest/src/rest.server.ts

```diff
@@ -979,20 +954,24 @@ export interface ApiExplorerOptions {
 }
 
 /**
- * Options for RestServer configuration
+ * RestServer options
  */
-export interface RestServerOptions {
+export type RestServerOptions = Partial<RestServerResolvedOptions>;
+
+export interface RestServerResolvedOptions {
```

Comment:
We can leave `RestServerOptions` unchanged and declare resolved options as `Required<RestServerOptions>`. See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html.

Or event make it `ReadonlyRequired`:
```ts
type ReadonlyRequired<T> = { +readonly [P in keyof T]-?: T[P] };  // Add `readonly` and remove `?`
```

Reply:
Nice, I was not aware of `Required` type.

Please note that not all fields are required in the resolved config. The following properties are remaining optional: `basePath`, `sequence`, `requestBodyParser`.

Conceptually, it makes more sense to me to use the resolved config as the source of truth:
- RestServer defines a type describing configuration it uses at runtime. The shape of the config object as required by various implementation bits.
- Then we define another interface describing options accepted by the server. This interface uses optional properties for fields that have a reasonable default.